enhancements/cvo/clusteroperator-resource-handling.md

```diff
@@ -0,0 +1,100 @@
+---
+title: clusteroperator-resource-handling
+authors:
+  - "@deads2k"
+reviewers:
+  - "@abhinavdahiya"
+  - "@sdodson"
+approvers:
+  - "@abhinavdahiya"
+  - "@derekwaynecarr"
+creation-date: 2019-12-11
+last-updated: 2019-12-11
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# ClusterOperator Resource Handling
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+`clusteroperators.config.openshift.io` are used to determine success of installation.
+They are also used to drive collection of debugging data from tools like `oc adm inspect` and `oc adm must-gather`.
+The `clusteroperator` resources in a payload should always be present, even before installation of particular
+operators to see which `clusteroperators` need to report in and to allow establishing `.status.relatedResources` before
+an operator pod runs.
+This is critical for debugging clusters that fail to install or fail to upgrade with new operators present.
+
+## Motivation
+
+Debugging information for clusters that succeed in bootstrapping, but fail during installation is missing most
+of the data required to resolve the issue via must-gather.
+
+### Goals
+
+1. allow collection of debugging data for failed installs using normal tools. 
+
+### Non-Goals
+
+1. create a new tool to gather data for failed installs after bootstrapping.
+
+## Proposal
+
+1. `clusteroperator` resources in the payload should be created with the required status conditions (available, progressing,
+   degraded) set to `Unknown`.
```

Comment:
https://github.com/openshift/api/blob/9ffd03c1c270ddd8cbb625295b1a74ade7e01229/config/v1/types_cluster_operator.go#L141-L175

has more than listed here, which ones do we care about now, and how we get this initial list updated when more conditions become _the norm_

Reply:
> https://github.com/openshift/api/blob/9ffd03c1c270ddd8cbb625295b1a74ade7e01229/config/v1/types_cluster_operator.go#L141-L175
> 
> has more than listed here, which ones do we care about now, and how we get this initial list updated when more conditions become _the norm_

Only those three are required. I would draw the line at required.